src/Promitor.Agents.Scraper/Validation/MetricDefinitions/MetricsValidator.cs

```diff
@@ -63,13 +63,16 @@ private IEnumerable<string> Validate(MetricDefinition metric)
                 errorMessages.AddRange(metricDefinitionValidationErrors);
             }
 
-            var metricAggregationValidator = new AzureMetricConfigurationValidator(_metricDefaults);
-            var metricsConfigurationErrorMessages = metricAggregationValidator.Validate(metric.AzureMetricConfiguration);
-            errorMessages.AddRange(metricsConfigurationErrorMessages);
+            if (metric.ResourceType != ResourceType.LogAnalytics)
```

Comment:
Hm, I'm not sure why we are skipping this part because the new Log Analytics scraper should just be another scraper and align with the existing approach?

Can you elaborate on this please?

Reply:
Because if resourceType = LogAnalytics then our metrics definition will have `logAnalyticsConfiguration`  field but `azureMetricConfiguration` will be null so I don't want to validate it here. 

But this way seem to be ugly.  I am thinking a better way to do this

Reply:
> metrics definition will have logAnalyticsConfiguration field but azureMetricConfiguration will be null so I don't validate it here.

AFAIK there was no validation for this? Can you add this please?

Also, it might be nicer if we add that validation inside the AzureMetricConfigurationValidator by passing the resource to it so that the logic is captured there maybe?

Reply:
Added some validation in AzureMetricConfigurationValidator